stdlib/asm/crypto/stark/utils.masm

```diff
@@ -17,35 +21,55 @@ export.compute_lde_generator
     # => [domain_gen, ..]
 end
 
-#! Check that the Proof-of-Work contained in the current `SEED` is equal to the required number
+#! Check that the Proof-of-Work contained in the nonce is equal to the required number
 #! of bits prescribed by grinding bits. Currently the grinding factor is fixed to 16 bits.
 #!
 #! Input: [...]
 #! Output: [...]
-#! Cycles: 42
+#! Cycles: 53
 export.check_pow
```

Comment:
I wonder if this actually should be a part of `random_coin` module as well. Maybe the signature could look like:
```
#! Input: [grinding_factor, ...]
#! Output: [...]
export.check_pow
```
If we do this, we might be able to get rid of state getters/setters from `random_coin` as I think this is the only place where they are used.

Reply:
Yep - that's more than sufficient.